src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -37,4 +56,286 @@ public void testDatabaseMetaDataWrapper() throws SQLException {
+            endIndex = connectionString.indexOf(";", startIndex);
+            startIndex = startIndex + "user=".length();
+        }
+
+        String userFromConnectionString = connectionString.substring(startIndex, endIndex);
+        String userName = databaseMetaData.getUserName();
+
+        assertNotNull(userName, "databaseMetaData.getUserName() should not be null");
+
+        assertTrue(userName.equalsIgnoreCase(userFromConnectionString),
+                "databaseMetaData.getUserName() should match with UserName from Connection String.");
+    }
+
+    /**
+     * Testing of {@link SQLServerDatabaseMetaData#getSchemas()}
+     * @throws SQLServerException
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchema() throws SQLServerException, SQLException {
+        DatabaseMetaData databaseMetaData = connection.getMetaData();
+
+        ResultSet rs = databaseMetaData.getSchemas();
+
+        while (rs.next()) {
+            assertTrue(!StringUtils.isEmpty(rs.getString(1)), "Schema Name should not Empty");
+        }
+    }
+
+    /**
+     * Get All Tables.
+     * 
+     * @throws SQLServerException
+     * @throws SQLException
+     */
+    @Test
+    public void testDBTables() throws SQLServerException, SQLException {
+        DatabaseMetaData databaseMetaData = connection.getMetaData();
+        
+        ResultSet rsCatalog = databaseMetaData.getCatalogs();
+        
+        assertTrue(rsCatalog.next(), "We should get atleast one catalog"); 
+        
+        String[] types = {"TABLE"};
+        ResultSet rs = databaseMetaData.getTables(rsCatalog.getString("TABLE_CAT"), null, "%", types);
+        
+        while (rs.next()) {
+            assertTrue(!StringUtils.isEmpty(rs.getString("TABLE_NAME")),"Table Name should not Empty"); 
+        }
+    }
+
+    /**
+     * Testing DB Columns.<p>
+     * We can Improve this test scenario by following way.
+     * <ul>
+     *  <li> Create table with appropriate column size, data types,auto increment, NULLABLE etc.  
+     *  <li> Then get databasemetatadata.getColumns to see if there is any mismatch.
+     * </ul>
+     * @throws SQLServerException
+     * @throws SQLException
+     */
+    @Test
+    public void testGetDBColumn() throws SQLServerException, SQLException {
+        DatabaseMetaData databaseMetaData = connection.getMetaData();
+        String[] types = {"TABLE"};
+        ResultSet rs = databaseMetaData.getTables(null, null, "%", types);
+        
+        //Fetch one table
+        assertTrue(rs.next(), "At least one table should be found");
+        
+        //Go through all columns.
+        ResultSet rs1 = databaseMetaData.getColumns(null, null, rs.getString("TABLE_NAME"), "%");
+        
+        while (rs1.next()) {
+            assertTrue(!StringUtils.isEmpty(rs1.getString("TABLE_CAT")),"Category Name should not Empty"); //1
+            assertTrue(!StringUtils.isEmpty(rs1.getString("TABLE_SCHEM")),"SCHEMA Name should not Empty");
```

Comment:
error message should be ...should not be Empty

Fix this for all the error messages